syncstorage/storage/sql/dbconnect.py

```diff
@@ -144,21 +159,85 @@ def _get_bso_columns(table_name):
 
 bso = Table("bso", metadata, *_get_bso_columns("bso"))
 
+# Table mapping (user_id, collection_id) => batch IDs
+
+batch_uploads = Table(
+    "batch_uploads",
+    metadata,
+    Column("batch", BigInteger, primary_key=True, nullable=False),
```

Comment:
D'oh!  It's supposed to be PRIMARY KEY (batch, userid).  Looks like this plus the autoincrement=False on the collection column are victims are failed copying and pasting.